### DISCUSSION TITLE
Option to remove the "generated from" header tag

### DISCUSSION THREAD

**Discussion by sshquack** [community/community#15900] [Other Feature Feedback, Questions, & Ideas]
# Problem
- Today there is no setting to remove the "generated from" repo tag.
- The "generated from" tag looks exactly like "forked from" so it looks unprofessional.

# Potential solution
Give users the option to remove the "Generated from <source_repo>" from the repo from the settings k.

![Screenshot 000217@2x](https://user-images.githubusercontent.com/5678459/166175916-17c23e01-089f-4289-91bb-c22760f49853.png)

# Around the web
- [This StackOverflow issue](https://stackoverflow.com/questions/66587174/how-to-remove-generated-from-tag) that has been viewed over 2000 times
- [This community post](https://github.community/t/remove-generated-from-header-on-repository-generated-from-template/146024) has been viewed 2700 times

**Comment by Vasile-Peste** [comment]
I'm wondering if it's at least possible to remove it by contacting GitHub support?

**Reply by hackx2** [reply]
sadly, it isn't.

**Comment by Basssiiie** [comment]
It would be great if it were possible to remove it. I have a repository that was generated from a template years ago, but it does not resemble the template anymore at all, making the "generated from" kinda misleading and a lie about the current situation.

**Comment by orhanerday** [comment]
+1

**Comment by technorav3nn** [comment]
+1

**Comment by irshadahmad21** [comment]
+1

**Comment by axelpale** [comment]
The templates should help people to spin up their project quickly. However, templates are relatively barebone and therefore should not dangle around forever if at all, unlike forks that add true informational value. At first, the template tag link may be useful by providing the developer a shortcut to access the template documentation, but after the project is set up, the template tag is just extra clutter.

Personally I feel the tag makes the project look a bit "cheap". Currently, to avoid this feel and look, I need to copy the template files manually to the new project, thus nullifying the purpose of the Template feature in the first place.

Please GitHub, just make it optional for the repo maintainer to show the tag, just in the way the maintainer can currently hide Issues, Wiki, and Projects.

**Comment by hjhjdev** [comment]
+1

**Comment by fiddlermikey** [comment]
+1
My devs are asking for this too

**Comment by pavan2108** [comment]
+1

**Reply by LISA-KOREA** [reply]
🙄

**Comment by stag-enterprises** [comment]
+1

**Comment by huipvandenende** [comment]
+1

**Comment by dmitrii-didenko** [comment]
+1

**Comment by sjkoelle** [comment]
+1

**Comment by jarahn** [comment]
It's a bit of an annoying workaround, but I was able to get rid of the "generated from" tag directly from the Github web UI. (Note: same as the solution in the stack overflow post, this will not preserve PRs, issues, etc.) Steps: 

- Click "use this template" and create a new repo from it as normal
- Mark the new repo (which contains the "generated from" tag) as a template repository in settings
- **From the repo you just created** click "use this template" and create another new repo
- Delete the first repo you created (parent of the one in the last step) and the tag should disappear from the child

**Reply by fiddlermikey** [reply]
That is an interesting workaround, and you're right, kind-of annoying too. lol This wouldn't be too bad for an individual dev, but not really good for most orgs with multiple devs. 

# A+ for creativity.

**Reply by aiadam36** [reply]
You've saved my whole career. Big Thanks!

**Reply by mrlubos** [reply]
This is what IQ140 looks like

**Reply by KanagawaMarcos** [reply]
You are THE MAN
<img width="500" height="500" alt="image" src="https://github.com/user-attachments/assets/6cb4b4dd-5ada-436c-b956-3e08a003be54" />

**Comment by drl990114** [comment]
I found another way, although maybe not perfect. If the template repo is your own, you can delete the template repo first, so that this tag will disappear.

**Reply by drl990114** [reply]
The current effective solution can be found here： https://stackoverflow.com/a/76436754/16346493

**Comment by mwt** [comment]
You can always git sync to a fresh repo!

**Comment by irshadahmad21** [comment]
All the replies mentioning to git sync/push to a fresh repo should understand that it will result in loosing all the issues/PRs/discussions etc.

**Reply by mwt** [reply]
Yeah, I think you might be able to get rid of it by making the repo private and then public again. Though, I think this has all the same caveats.

**Reply by irshadahmad21** [reply]
No, it doesn't help.

**Comment by thomazmz** [comment]
I was unpleasantly surprised when I realized repos created from templates would have a permanent remark linking to the template that originated then. The fact that the information is available in such an explicit/highlighted way (as if something incredibly relevant for daily work), just makes things even worse.

As previously mentioned by others in this thread, templates are widely expected to be a starting point for a code base, but not a relevant thing to be taken into consideration in the project's long run. Thus, not something that is worth mentioning in such a way. It should be a shortcut for repository creation, not a long-lived aspect of the project it originates.

It is different from a fork: When you fork a project you usually benefit from having that fork moment explicitly noted in your repos's history; contributors then have a contextualized reference to understand your project's past and also pick into what the original forked project has turned afterwards. 

Do agree it is nice to keep commit history coming from templates, but not keeping it could also be justifiable. It is then just a matter of offering users an option to "ignore template commits". If the option is marked then you just create a new repo with a general commit that squashes the whole template once. No mention to the template needed further on in any place of the originated repo - at all. 

Consider the case where a repo is created keeping the whole commit history, then the reference to the original one might be worth keeping somewhere, but not in such a place where it is today. This could probably live somewhere under repository settings, and then an option to display it on the project's main page could be provided (which would then be unchecked and the owner would have to opt in).

Also, bear in mind that the template itself might also change a lot in the long run, so merely linking it to the template's main page is worth nothing. if there will be a clickable link somewhere it should probably redirect the user to the point in the template history that originated the repo.

**Comment by chrisj-au** [comment]
Why on earth would you want a permanent reminder like that?  For something that would take no time to at least disable from displaying I am pretty disappointed GH have had nothing to say since this was raised over a year ago.

**Comment by jharlow** [comment]
+1

**Comment by caiocinel** [comment]
+1

**Comment by emilhorlyck** [comment]
+1

**Comment by martinyonatann** [comment]
+1

**Comment by orhanerday** [comment]
+12323879127917839879237827813287721712789

**Reply by 0392272713** [reply]
nam_van3@icloud.com

**Comment by 949886** [comment]
+1

**Reply by 0392272713** [reply]
nam_van3@icloud.com

**Comment by eaguad1337** [comment]
+1

**Comment by 0392272713** [comment]
nam_van3@icloud.com

**Comment by mpwang** [comment]
+1

**Comment by stephendotgg** [comment]
+1

**Comment by avasilic** [comment]
+1

**Comment by adamking** [comment]
+1

**Comment by hprotzek** [comment]
+1

**Comment by Emilstromberg** [comment]
+1

**Comment by adstr123** [comment]
+1

**Comment by mofeejegi** [comment]
+1

**Comment by ClementRoyer** [comment]
+1

**Comment by jamtur01** [comment]
+1

**Comment by TBM13** [comment]
+1

**Comment by kee-oth** [comment]
+1

**Comment by billyjacoby** [comment]
+1

**Comment by tsachev** [comment]
+1

**Comment by bcc829** [comment]
+1

**Comment by jenseva** [comment]
+1

**Comment by linxianxi** [comment]
+1

**Comment by noxware** [comment]
+1

**Comment by dereckmezquita** [comment]
+1

**Comment by doraemonxxx** [comment]
`+1`

**Comment by Coffee-Nerd** [comment]
+1 pleaaaase

**Comment by elloboblanco** [comment]
Would love this 🙌🏻

**Comment by AbitTheGray** [comment]
+1

I use Templates for all C++ projects as I have everything configured there (CMake project, tests, Github Actions...) and it does not matter what IDE or PC I use.
So it is unnecessary fluff in all my C++ repositories. I even considered deleting those repos and creating new ones just so I won't have the text there.

My suggestion - add toggle to make the text visible (in repository settings and/or to the gear icon next to "About" header).
This way it will be possible to bring it back (in case you want/need it for something) but not change the value.

**Comment by CoryCharlton** [comment]
+1

As a workaround I found that setting my template repos to private hides the "generated from" header to other users that do not have access to the private template repo.

**Comment by shayanb** [comment]
+1

**Comment by mkannwischer** [comment]
+1

**Comment by t-schreibs** [comment]
+1

**Comment by davidnknight** [comment]
+1

**Comment by brettclanton001** [comment]
+1

**Comment by fgaletic** [comment]
+1

**Comment by alexander-roemer** [comment]
+1

**Comment by red2678** [comment]
+1

**Reply by proyacerta** [reply]
+1

**Comment by anieblas** [comment]
+1

**Comment by DiegoAndresCortes** [comment]
+1

**Comment by AFMiziara** [comment]
+1

**Comment by kentbull** [comment]
+1

**Comment by INFCode** [comment]
+1

**Comment by tiagosiebler** [comment]
+1

**Comment by dkirkby** [comment]
+1

**Comment by leclairm** [comment]
I just don't get why removing this tag still isn't an option.
We have a web site generated by mkdocs that still mentions "generated from https://github.com/just-the-docs/just-the-docs-template". This is insane. How much effort does it require from the zillions of dev in the GH company bought by the Microsoft company to add this ridiculously tiny feature, yet required by so many users since 2 years??

**Reply by janderssonse** [reply]
They have a bunch of issues like this att GitHub - AKA small fixes that would add a lot of daily value to thousands of users, but seems ignored since years. I wonder if someone at GitHub reads these community discussions

**Comment by Hronom** [comment]
More on this it also reflected in API, so if you use terraform to manage GitHub repository you will get it in state of it.

I think it's not minor and should be done. And not seat here years

**Comment by uen** [comment]
I don't understand why anyone would ever use a template if the repository will be permanently branded with the template name..?

**Comment by daveycodez** [comment]
HELLOOOOO GITHUB

**Reply by leclairm** [reply]
"You have reached the voicemail of GitHub, please leave a message after the tone. There is a non zero chance that we will acknowledge receipt"

**Comment by alexdovzhanyn** [comment]
+1 LGTM

**Comment by fgaletic** [comment]
Jesus yes

**Comment by alexdovzhanyn** [comment]
For now, a workaround is to create a new repository, then change your origin in your original repo locally to point to the new one. then, once you push, you'll keep all the commit history and everything but the "generated from" tag will be gone.

Github still needs to add this simple feature though so we don't have to jump through these silly hoops to get things set up how we want.

**Reply by felipecrs** [reply]
I suppose you lose stars, issues, PRs, by doing that, right?

**Reply by alexdovzhanyn** [reply]
Unfortunately, yes -- that's a good call out.

**Reply by daveycodez** [reply]
The other awkward solution is to make the Repo you made it a template from Private, then it won't show to anyone who doesn't have access to the OG template. Or delete the OG template rofl. Make a template of the template, then make the repo from the template template, then delete the template template. Am I sane?

**Reply by felipecrs** [reply]
No.

**Comment by grantespo** [comment]
+1

**Comment by Sup2point0** [comment]
+0!

**Comment by sj-ganwh** [comment]
+1

**Comment by qdequippe** [comment]
+1

**Comment by seyfer** [comment]
+1

**Comment by HyperPaint** [comment]
+1

**Comment by MediaMarco** [comment]
+1

**Comment by numberisnan** [comment]
+1

Seems silly this isn't a feature. Attribution is cool, but I'm doing a total rewrite!

**Comment by chdeskur** [comment]
+1

**Comment by Lermatroid** [comment]
+1

Attribution is important though! Maybe have it where when its deleted it moves somewhere in the sidebar of the repo? Just the current header is very prominent.

**Comment by sebacarrasco93** [comment]
+1

**Comment by arshad-yaseen** [comment]
+1

**Comment by zyyv** [comment]
+1

**Comment by tdwesten** [comment]
+1

**Comment by davidweiss** [comment]
+1

**Comment by hackx2** [comment]
+1

**Comment by Erubinyl** [comment]
+1

**Comment by prpkng** [comment]
+1

**Comment by AndiKod** [comment]
So far, as Github seems not in a hurry with addressing that specific point, the simple preventive steps still working well is again:

- Don't use the "Use this Template" button to create a repo
- Clone it directly, create a blank repo and replace the origin remote
- You end up with the same starting point, but no "Generated with X"
- Giving initial credit to the Template is still possible in the README

**Reply by VelocityRa** [reply]
This is not enough for some templates that contain post-processing via GitHub Actions, like https://github.com/cpp-best-practices/cmake_template.
In that case you need to create a temporary repo, get the contents and paste them in your repo.

**Comment by bennettldavid** [comment]
+1

**Comment by vovchisko** [comment]
+1

**Comment by seltinusjulio** [comment]
+1

**Comment by IevgenTarasov** [comment]
+1

Funny that if you ask Github Copilot, it hallucinates that there's an option to detach from template in Repo Settings

**Comment by MatheusFS-dev** [comment]
+1

**Comment by BenCos17** [comment]
why is this still an issue GitHub....
I wouldn't even mind if it was less prominent but it's literally in full view of everyone who views it

**Comment by sQverful** [comment]
+1

**Comment by mjyb16** [comment]
+1

**Comment by medyagh** [comment]
+1

**Comment by PedroRomanoBarbosa** [comment]
+1

**Comment by Guation** [comment]
+1

**Comment by fan-ziqi** [comment]
+1

**Comment by thebeachdev** [comment]
+1

**Comment by daveycodez** [comment]
ASFSDFSAFDSFASDFASFGDAFGADSFGsafg

**Comment by florianamette** [comment]
+1

**Comment by anthonioxzxeroz-cpu** [comment]
Enviado desde mi iPhoneEl 21/11/2025, a la(s) 17:33, F4K ***@***.***> escribió:﻿
+1

**Comment by matildasmets** [comment]
+1

**Comment by trsonic** [comment]
+1

**Comment by xevrion** [comment]
+1

**Comment by powerhousedev-pedro** [comment]
+1

**Comment by calvinsnax** [comment]
+1

**Comment by smith558** [comment]
+1

**Comment by tgies** [comment]
+1

**Comment by arbabkhalil** [comment]
+1

**Comment by Tokisaki-Galaxy** [comment]
+1, I need it

**Comment by HagaSpa** [comment]
+1

**Comment by connorchewning** [comment]
+1

**Comment by adeelzaman** [comment]
+1

**Comment by harrisonk0** [comment]
+1

**Comment by helen** [comment]
Hello! We've moved this indicator to the bottom of the repo home sidebar / about info. Still considering an option, mostly about where to put it for discoverability and not overloading settings even further, but hope that this is at least a small improvement in the meantime!

**Reply by felipecrs** [reply]
It's certainly a great improvement. It doesn't bother anymore nearly as much. Thanks!

**Reply by TokisanGames** [reply]
Thank you for moving it. Since you're still considering keeping the option, please open it up for discussion as we (the sufferers of your choices) have more experiences to provide. For probably most of the people on this thread, the option is irrelevant. That is my case, as the project I generated from was just a template one of my team members used which has contributed only a basic build script. For years we had that annoying and distracting tag as the number 2 line in our repo, and no option to remove it without rebuilding our repo at a great cost of losing data.

**Reply by Basssiiie** [reply]
What a time to be alive! Thanks! 🚀

**Comment by uen** [comment]
4 years to consider that
